File: test_data/sample_code.py
# 示例代码文件 - 包含一些需要审查和修复的问题

def calculate_total(items):
    """计算总价"""
    total = 0
    for item in items:
        total += item['price']  # 缺少数量检查
    return total

def process_user_input(user_input):
    """处理用户输入 - 存在安全问题"""
    # 安全问题：直接执行用户输入
    result = eval(user_input)
    return result

def fetch_data(url):
    """获取数据 - 缺少错误处理"""
    import urllib.request
    response = urllib.request.urlopen(url)
    return response.read()

class UserManager:
    """用户管理器"""
    
    def __init__(self):
        self.users = []
    
    def add_user(self, username, password):
        """添加用户 - 密码明文存储"""
        self.users.append({
            'username': username,
            'password': password  # 安全问题：密码应该加密
        })
    
    def authenticate(self, username, password):
        """认证用户"""
        for user in self.users:
            if user['username'] == username and user['password'] == password:
                return True
        return False

File: test_data/sample_code.js
// 示例 JavaScript 代码 - 包含一些需要审查的问题

function calculateTotal(items) {
    // 性能问题：没有使用 reduce
    let total = 0;
    for (let i = 0; i < items.length; i++) {
        total += items[i].price;
    }
    return total;
}

function processUserInput(userInput) {
    // 安全问题：使用 eval
    return eval(userInput);
}

async function fetchData(url) {
    // 缺少错误处理
    const response = await fetch(url);
    return await response.json();
}

class UserManager {
    constructor() {
        this.users = [];
    }
    
    addUser(username, password) {
        // 安全问题：密码明文存储
        this.users.push({
            username: username,
            password: password
        });
    }
    
    authenticate(username, password) {
        // 性能问题：线性搜索
        for (let user of this.users) {
            if (user.username === username && user.password === password) {
                return true;
            }
        }
        return false;
    }
}
